src/julia_gc.cc

```diff
@@ -703,28 +988,33 @@ static void GapTaskScanner(jl_task_t * task, int root_task)
     size_t size;
     int    tid;
     char * stack = (char *)jl_task_stack_buffer(task, &size, &tid);
+    // If it is the current task, it has been scanned by GapRootScanner()
+    // already.
+    if (task == JuliaTLS->current_task)
+        return;
+    int rescan = 1;
+    if (!FullGC) {
+        // This is a temp hack to work around a problem with the
+        // generational GC. Basically, task stacks are being scanned
+        // regardless of whether they are old or new. In order to avoid
```

Comment:
Is this a bug in Julia, resp. something there we don't understand? If so, can you please provide a link to the relevant Julia issue report, resp. the relevant discussion on the Julia discourse?

Reply:
I'm not sure if its a bug or an inefficiency. Basically, as I understand it, the task stacks can be roots themselves without being referenced from elsewhere. So, in parts, this is most likely intentional and simply hasn't created any performance issues, as task stacks tend to be short.